lib/ansible/plugins/action/__init__.py

```diff
@@ -568,7 +571,9 @@ def _remote_expand_user(self, path, sudoable=True, pathsep=None):
         expand_path = split_path[0]
 
         if expand_path == '~':
-            if sudoable and self._play_context.become and self._play_context.become_user:
+            if self._connection.remote_is_local:
```

Comment:
Given the conditional on line 249 would we ever reach this, or have you just added this in case in the future their is a `self._remote_expand_user` call?

Reply:
We won't hit it for remote_tmp, but we probably should never try to expand remote_user either

Reply:
Move this condition up a level:
``` python
if expand_path == '~' and not self._connection.remote_is_local:
```